### ISSUE TITLE
[Bug] PDF generation fails with NullReferenceException and creates corrupt pdf file

### ISSUE THREAD

**Issue by Lagann311** [dotnet/docfx#8987]
**Describe the bug**
When running **docfx pdf**, an error occurs after printing pages is done:

Resolving links (4/6)
Loading headers and footers (5/6)
Printing pages (6/6)
Done
    Error happen when converting pdf/toc.json to Pdf. Details: System.NullReferenceException: Object reference not set to an instance 
    of an object.
    at UglyToad.PdfPig.Writer.PdfDocumentBuilder.CompleteDocument()
    at UglyToad.PdfPig.Writer.PdfDocumentBuilder.Dispose()
    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.WriteOutlines (MemoryStream input, Stream output) in 
    /_/src/Microsoft.DocAsCode.HtmlToPdf/HtmlToPdfConverter.cs:line 301
    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.SaveCore (Stream stream) in 
    /_/src/Microsoft.DocAsCode.HtmlToPdf/HtmlToPdfConverter.cs:line 267
    at Microsoft.DocAsCode.HtmlToPdf.HtmlToPdfConverter.Save(String outputFileName) in 
    /_/src/Microsoft.DocAsCode.HtmlToPdf/HtmlToPdfConverter.cs:line 91
    at Microsoft.DocAsCode.HtmlToPdf.ConvertWrapper.ConvertCore (String basePath, String pdfFileName, IList`1 htmlModels) in 
    /_/src/Microsoft.DocAsCode.HtmlToPdf/ConvertWrapper.cs:line 342
    at Microsoft.DocAsCode.HtmlToPdf.ConvertWrapper.<>c__DisplayClass7_0.<ConvertCore>b___1(ManifestItem tocFile) in 
    /_/src/Microsoft.DocAsCode.HtmlToPdf/ConvertWrapper.cs:line 166 
Completed Scope: PDF in 14729,6153 milliseconds.

Build failed.

**To Reproduce**
1. Create a new docfx project with supplied docfx.json (example project needs to be added)
2. Make sure wkhtmltopdf is installed
3. Run **docfx pdf**

**Expected behavior**
PDF generation process finishes without errors and creates valid pdf file

**Context (please complete the following information):**
- OS: Windows 10
- Docfx version: DocFX 2.67.5

**Additional context**
[docfx.json.txt](https://github.com/dotnet/docfx/files/12081154/docfx.json.txt)

This wasn't an issue with earlier versions like 2.59.4.0

**Comment by Lagann311** [comment]
I made some changes to my docfx.json file and the error is still showing up, but at least the PDF file generation works as expected now and no longer creates a corrupt file:

```
  "metadata": [
    {
      "src": [
        {
          "files": [
            "*.csproj"
          ],
          "src": "Path/to/.csproj/containing/folder"
        }
      ]
    }
  ]
```

```
  "pdf": {
    "content": [
      {
        "files": [ "**/*.{md,yml}" ]
      }
    ]
  }
```

[docfx.json_new.txt](https://github.com/dotnet/docfx/files/12122155/docfx.json_new.txt)

**Comment by yufeih** [comment]
Is there a minimal repo project publicly available?

**Comment by Lagann311** [comment]
I'm not able to supply a repo project unfortunately.

**Comment by gitsourabhmishranotes** [comment]
Hi,

I also get the same problem, and finally i got the solution for this. Regarding this problem two software version match is required
1. DocFx
2. wkhtmltopdf
So regarding this problem, this is the problem of version mismatch. you have to install 
1. docfx 2.63.0+9a8a2abc13c595472cac70121177ed5f9f55237e
2. wkhtmltox-0.12.6-1.msvc2015-win64

These two combination versions will solve your problem. 

Thanks and Regards
Sourabh Mishra Notes

**Comment by filzrev** [comment]
It seems `NullReferenceException` is occurs when specifying `outline` settings.
> "outline": "WkDefaultOutline",

https://github.com/dotnet/docfx/blob/b5ac57da4501fe9aee50c39120e983f6275ec90a/src/Docfx.HtmlToPdf/HtmlToPdfConverter.cs#L223-L225

 Error is disappeared by modifying above line to return `Array.Empty<BookmarkNode>();` instead of `null`.

I've tested on following environment.
- OS: Windows
- docfx version: latest source
- project: `docfx\samples\seed`

**Comment by yufeih** [comment]
Fixed in v2.73.0 with [a new PDF engine](https://dotnet.github.io/docfx/docs/pdf.html).
